src/lib/serviceRoutinesV2/getAllSubscriptions.cpp

```diff
@@ -87,7 +87,11 @@ std::string getAllSubscriptions
   }
 
   std::string out;
+
+  if (limit != 0)
+  {
   TIMED_RENDER(out = vectorToJson(subs));
+  }
```

Comment:
You shouldn't do the limit=0 check here here. This is probably the cause of the problem in .test described in [previous comment](https://github.com/telefonicaid/fiware-orion/pull/3994#discussion_r783055063).

Instead, you should deal with the limit=0 condition internally at mongoListSubscriptions() (or even in collectionRangedQuery()).

Reply:
Added condition at mongoListSubscriptions()